src/backend/joanie/tests/core/api/order/test_read_detail.py

```diff
@@ -137,3 +139,110 @@ def test_api_order_read_detail_authenticated_not_owner(self):
+    )
+    def test_api_order_read_detail_authenticated_organization_owner(
+        self, _mock_thumbnail
+    ):
+        """Organization owners should be allowed to retrieve an order."""
+        organization_owner = factories.UserFactory()
+        organization = factories.OrganizationFactory()
+        factories.UserOrganizationAccessFactory(
+            user=organization_owner, organization=organization, role=enums.OWNER
+        )
+        *target_courses, _other_course = factories.CourseFactory.create_batch(
+            3, organizations=[organization]
+        )
+        product = factories.ProductFactory(target_courses=target_courses)
+        course = target_courses[0]
+        factories.CourseProductRelationFactory(
+            course=course, product=product, organizations=[organization]
+        )
+        order = factories.OrderFactory(
+            course=course, product=product, organization=organization
+        )
+        token = self.generate_token_from_user(organization_owner)
+
+        with self.assertNumQueries(7):
+            response = self.client.get(
+                f"/api/v1.0/orders/{order.id}/",
+                HTTP_AUTHORIZATION=f"Bearer {token}",
+            )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertDictEqual(
+            response.json(),
+            {
+                "id": str(order.id),
+                "certificate_id": None,
+                "contract": None,
+                "course": {
+                    "code": order.course.code,
+                    "id": str(order.course.id),
+                    "title": order.course.title,
+                    "cover": "_this_field_is_mocked",
+                },
+                "created_on": order.created_on.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
+                "enrollment": None,
+                "state": order.state,
+                "main_invoice_reference": None,
+                "order_group_id": None,
+                "organization_id": str(order.organization.id),
+                "owner": order.owner.username,
+                "total": float(product.price),
+                "total_currency": settings.DEFAULT_CURRENCY,
+                "product_id": str(product.id),
+                "target_enrollments": [],
+                "target_courses": [
+                    {
+                        "code": target_course.code,
+                        "course_runs": [
+                            {
+                                "id": course_run.id,
+                                "title": course_run.title,
+                                "resource_link": course_run.resource_link,
+                                "state": {
+                                    "priority": course_run.state["priority"],
+                                    "datetime": course_run.state["datetime"]
+                                    .isoformat()
+                                    .replace("+00:00", "Z"),
```

Comment:
I'm surprised you didn't create yet a helper doing this.

Reply:
You know me so well :heart_eyes: